SQL/DDL/data-types/handling-numbers.md

```diff
@@ -0,0 +1,95 @@
+# Handling Numbers
+author: mihaiberq
+
+levels:
+
+  - beginner
+
+  - basic
+
+type: normal
+
+category: must-know
+
+tags:
+  - introduction
+  - workout
+
+
+---
+## Content
+
+Probably the most used datatypes in tables are **numeric values**. They can be used for *id*s, *stats*, *attack points* etc. Even **true/false** (`bool`s or `boolean`s) values are considered to be numeric types.
+
+Most SQL implementations (PostgreSQL, MySQL, Microsoft SQL) have numeric types with the same capabilities, but under different names.
+
+It is worth noting that, while MySQL and MSSQL support both signed and unsigned values, PostgreSQL accepts only unsigned values.
+
+### Bits
+
+A string of bits, up to 64 bits long:
+- `bit(n)`: where *n* is the maximum length (PSQL, MYSQL)
+
+### Integers
+
+Called either `int` or `integer`, this data type accepts 32bit/4 bytes integers in all SQL implementations. In terms of range, there are some differences:
+- unsigned (PSQL, MySQL, MSSQL): `0` to `2^32 - 1` (`0` to `4,294,967,295`)
+- signed (MySQL, MSSQL): `-2^31` to `2^31 - 1` (`-2,147,483,648` to `2,147,483,647`)
+
+There are also prefixes for cases when you know the values won't reach a certain bound, which helps mitigating the memory footprint.
+
+Smaller integers are as follows:
+- `tinyint`: 8bit/1 bytes (MySQL, MSSQL)
+- `smallint`: 16bit/2 bytes (PSQL, MySQL, MSSQL)
+
+Bigger integers are as follows:
+- `bigint`: 64bit/8 bytes (PSQL, MySQL, MSSQL)
+
+### Serials
+
+Available only in PSQL and MySQL, serials are unsigned auto-incrementing integers. MySQL's `serial` defaults to 8 bytes values (`bigint`), while in PSQL all three variants are available (`smallserial`, `serial`, `bigserial`).
```

Comment:
Add a note that if you are using Serial as IDs, use `bigint`, with a footnote to this link:
https://hackernoon.com/the-night-the-postgresql-ids-ran-out-9430a2dbb895